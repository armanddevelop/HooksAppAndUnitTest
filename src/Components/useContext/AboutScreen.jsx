import { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  const userRest = {};
  return (
    <div>
      <h1>AboutScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null)}</pre>
      <button className="btn btn-warning" onClick={() => setUser(userRest)}>
        Log out
      </button>
    </div>
  );
};
