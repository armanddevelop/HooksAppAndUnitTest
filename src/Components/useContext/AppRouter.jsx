import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { LognScreen, AboutScreen, HomeScreen, NavBar } from "./index";
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Routes>
            <Route exact path="/" element={<HomeScreen />} />
            <Route exact path="/about" element={<AboutScreen />} />
            <Route exact path="/login" element={<LognScreen />} />
            <Route path="*" element={<HomeScreen />} />
          </Routes>
        </div>
      </div>
    </Router>
  );
};
