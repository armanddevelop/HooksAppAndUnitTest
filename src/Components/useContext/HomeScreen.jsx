import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  //console.log("esto vale userContext ", user);
  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null)}</pre>
    </div>
  );
};
