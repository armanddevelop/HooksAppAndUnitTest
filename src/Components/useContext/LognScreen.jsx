import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LognScreen = () => {
  const { setUser } = useContext(UserContext);
  const userObj = {
    id: 123,
    name: "creting",
  };
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(userObj)}>
        User
      </button>
    </div>
  );
};
