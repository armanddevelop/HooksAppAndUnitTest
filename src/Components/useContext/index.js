export { HomeScreen } from "./HomeScreen";
export { AboutScreen } from "./AboutScreen";
export { LognScreen } from "./LognScreen";
export { NavBar } from "./NavBar";
