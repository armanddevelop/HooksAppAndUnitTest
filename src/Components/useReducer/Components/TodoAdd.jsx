import React from "react";
import { useForm } from "../../../Hooks/useForm";
import PropTypes from "prop-types";
export const TodoAdd = ({ handleAddToDo }) => {
  const { formState, handleInputChange, resetField } = useForm({
    description: "",
  });
  const { description } = formState;
  const handleSubmit = (e) => {
    e.preventDefault();
    if (description.trim().length === 0) return;
    const newTodo = {
      id: new Date().getTime(),
      description,
      done: false,
    };
    handleAddToDo(newTodo);
    resetField();
  };
  return (
    <>
      <h4>Add Task</h4>
      <hr />
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          name="description"
          placeholder="learn code"
          autoComplete="off"
          className="form-control"
          value={description}
          onChange={handleInputChange}
        ></input>
        <button
          type="submit"
          className="btn btn-outline-primary mt-1 btn-block"
        >
          ADD
        </button>
      </form>
    </>
  );
};
TodoAdd.propTypes = {
  handleAddToDo: PropTypes.func.isRequired,
};
