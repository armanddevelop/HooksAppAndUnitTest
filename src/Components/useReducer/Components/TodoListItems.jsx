import PropTypes from "prop-types";
export const TodoListItems = ({
  description,
  done,
  idx,
  id,
  handleDelete,
  handleToogle,
}) => {
  return (
    <li className="list-group-item">
      <p
        className={`${done ? "complete" : ""}`}
        onClick={() => handleToogle(id)}
      >
        {idx + 1}.- {description}
      </p>
      <button onClick={() => handleDelete(id)} className="btn btn-danger">
        Delete
      </button>
    </li>
  );
};
TodoListItems.prototype = {
  description: PropTypes.string.isRequired,
  done: PropTypes.bool.isRequired,
  id: PropTypes.number.isRequired,
  idx: PropTypes.number.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToogle: PropTypes.func.isRequired,
};
