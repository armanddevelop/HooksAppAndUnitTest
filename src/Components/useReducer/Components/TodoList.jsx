import { TodoListItems } from "./TodoListItems";
import PropTypes from "prop-types";

export const TodoList = ({ todos, handleDelete, handleToggle }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map(({ id, description, done }, idx) => {
        return (
          <TodoListItems
            key={id}
            description={description}
            done={done}
            idx={idx}
            id={id}
            handleDelete={handleDelete}
            handleToogle={handleToggle}
          />
        );
      })}
    </ul>
  );
};
TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToggle: PropTypes.func.isRequired,
};
