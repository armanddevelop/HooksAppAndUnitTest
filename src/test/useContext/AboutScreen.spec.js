import { mount } from "enzyme";
import "@testing-library/jest-dom";
import toJson from "enzyme-to-json";
import { AboutScreen } from "../../Components/useContext";
import { UserContext } from "../../Components/useContext/UserContext";

describe("Testo to AboutScreen component", () => {
  const user = { name: "alicha", email: "euche@test.com" };
  const setUser = jest.fn();
  const container = mount(
    <UserContext.Provider value={{ user, setUser }}>
      <AboutScreen />
    </UserContext.Provider>
  );
  it("snapshot to AboutScreen", () => {
    expect(toJson(container)).toMatchSnapshot();
  });

  it("Delete the user object information in component", () => {
    container.find("button").simulate("click", () => {
      setUser();
    });
    expect(setUser).toHaveBeenCalledWith({});
  });
});
