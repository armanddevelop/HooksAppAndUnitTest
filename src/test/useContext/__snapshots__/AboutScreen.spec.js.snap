// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testo to AboutScreen component snapshot to AboutScreen 1`] = `
<AboutScreen>
  <div>
    <h1>
      AboutScreen
    </h1>
    <hr />
    <pre>
      {"name":"alicha","email":"euche@test.com"}
    </pre>
    <button
      className="btn btn-warning"
      onClick={[Function]}
    >
      Log out
    </button>
  </div>
</AboutScreen>
`;
